### PR TITLE
Add ANN Training Program using PyTorch

### PR DESCRIPTION
**Pull request by leonking1990**
## Description:
  This PR introduces the initial implementation of the Artificial Neural Network (ANN) training program for diabetes detection using PyTorch. It includes the following features:
  
  - ✅ AnnModel.py – Defines the ANN model with ReLU activations and a Sigmoid output for binary classification.
  - ✅ DataBuilder.py – Handles dataset loading, preprocessing (normalization), and DataLoader setup.
  - ✅ Integration with KaggleHub – Downloads the diabetes dataset dynamically, allowing customization of the download path.
  - ✅ Training & Evaluation – Implements training with binary cross-entropy loss and Adam optimizer, along with model evaluation.

## Changes Made:
  1. Implemented DiabetesModel – A fully connected feedforward neural network.
  2. Created DataBuilder – Automates data processing and loading.
  3. Ensured compatibility with CUDA – Uses GPU if available.
  4. Improved training script – Includes proper dataset splitting (80/20) and logging of loss per epoch.

## How to Test:
  1. Clone the repository and switch to the correct branch:
     ```sh
     git checkout AnnTraining
     ```
  2. Install dependencies:
      ```sh
      pip install torch pandas scikit-learn kagglehub
      ```
  3. Run the training script:
      ```sh
      python AnnModel.py
      ```
  4. Confirm that the model trains successfully and outputs loss per epoch.

## To-Do (Future Improvements):
   - Add hyperparameter tuning for learning rate and batch size.
   - Implement early stopping to prevent overfitting.
   - Save and load trained models for inference.

## Reviewers:
  - Please verify that the dataset downloads and loads correctly.
  - Check if the training process runs without errors.
  - Suggestions for improving model performance are welcome!

### CODE REVIEW

**Review by Cgfranco1 — APPROVED** [review]

**Review by babab0uille — APPROVED** [review]
Good Job!
